exchanges/kucoin/kucoin.go

```diff
@@ -311,352 +205,213 @@ func (ku *Kucoin) GetKlines(ctx context.Context, pair, period string, start, end
-		if err != nil {
-			return nil, err
-		}
-		klines[i].Volume, err = strconv.ParseFloat(resp[i][5], 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].Amount, err = strconv.ParseFloat(resp[i][6], 64)
-		if err != nil {
-			return nil, err
-		}
+		klines[i].StartTime = time.Unix(resp[i][0].Int64(), 0)
+		klines[i].Open = resp[i][1].Float64()
+		klines[i].Close = resp[i][2].Float64()
+		klines[i].High = resp[i][3].Float64()
+		klines[i].Low = resp[i][4].Float64()
+		klines[i].Volume = resp[i][5].Float64()
+		klines[i].Amount = resp[i][6].Float64()
 	}
 	return klines, nil
 }
 
-// GetCurrencies gets list of currencies
-func (ku *Kucoin) GetCurrencies(ctx context.Context) ([]Currency, error) {
-	var resp []Currency
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, kucoinGetCurrencies, &resp)
+// GetCurrenciesV3 the V3 of retrieving list of currencies
+func (ku *Kucoin) GetCurrenciesV3(ctx context.Context) ([]CurrencyDetail, error) {
+	var resp []CurrencyDetail
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, spotCurrenciesV3EPL, "/v3/currencies", &resp)
 }
 
-// GetCurrencyDetail gets currency detail using currency code and chain information.
-func (ku *Kucoin) GetCurrencyDetail(ctx context.Context, ccy, chain string) (*CurrencyDetail, error) {
-	if ccy == "" {
+// GetCurrencyDetailV3 V3 endpoint to gets currency detail using currency code and chain information.
+func (ku *Kucoin) GetCurrencyDetailV3(ctx context.Context, ccy currency.Code, chain string) (*CurrencyDetail, error) {
+	if ccy.IsEmpty() {
 		return nil, currency.ErrCurrencyCodeEmpty
 	}
 	params := url.Values{}
 	if chain != "" {
 		params.Set("chain", chain)
 	}
 	var resp *CurrencyDetail
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, common.EncodeURLValues(kucoinGetCurrency+strings.ToUpper(ccy), params), &resp)
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, spotCurrencyDetailEPL, common.EncodeURLValues("/v3/currencies/"+ccy.Upper().String(), params), &resp)
 }
 
 // GetFiatPrice gets fiat prices of currencies, default base currency is USD
-func (ku *Kucoin) GetFiatPrice(ctx context.Context, base, currencies string) (map[string]string, error) {
+func (ku *Kucoin) GetFiatPrice(ctx context.Context, base, currencies string) (map[string]types.Number, error) {
 	params := url.Values{}
 	if base != "" {
 		params.Set("base", base)
 	}
 	if currencies != "" {
 		params.Set("currencies", currencies)
 	}
-	var resp map[string]string
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, common.EncodeURLValues(kucoinGetFiatPrice, params), &resp)
+	var resp map[string]types.Number
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, fiatPriceEPL, common.EncodeURLValues("/v1/prices", params), &resp)
+}
+
+// GetLeveragedTokenInfo returns leveraged token information
+func (ku *Kucoin) GetLeveragedTokenInfo(ctx context.Context, ccy currency.Code) ([]LeveragedTokenInfo, error) {
```

Comment:
With this: https://www.kucoin.com/docs/rest/margin-trading/margin-info/get-leveraged-token-info
I would request you add `issuedSize`